torch_frame/data/multi_nested_tensor.py

```diff
@@ -120,83 +120,93 @@ def __repr__(self) -> str:
     def __getitem__(
         self,
         index: Any,
     ) -> Union['MultiNestedTensor', Tensor]:
+
         if isinstance(index, tuple):
-            # Get an element of (i, j). Returns Tensor
+            # index[0] for row indexing, index[1] for column indexing
             assert len(index) == 2
-            idx_0 = index[0]
-            # Support negative indices
-            if idx_0 < 0:
-                idx_0 = self.num_rows + idx_0
-            if idx_0 < 0 or idx_0 >= self.num_rows:
-                raise IndexError("Index out of bounds!")
-            idx_1 = index[1]
-            if idx_1 < 0:
-                idx_1 = self.num_cols + idx_1
-            if idx_1 < 0 or idx_1 >= self.num_cols:
-                raise IndexError("Index out of bounds!")
-
-            idx = idx_0 * self.num_cols + idx_1
-            start_idx = self.offset[idx]
-            end_idx = self.offset[idx + 1]
-            out = self.values[start_idx:end_idx]
-            return out
-
-        elif isinstance(index, int):
-            # Support negative indices
-            if index < 0:
-                index = self.num_rows + index
-            if index < 0 or index >= self.num_rows:
-                raise IndexError("Index out of bounds!")
-            # Get i-th row. Returns MultiNestedTensor
-            start_idx = index * self.num_cols
-            end_idx = (index + 1) * self.num_cols + 1
-            offset = self.offset[start_idx:end_idx]
-            values = self.values[offset[0]:offset[-1]]
-            offset = offset - offset[0]
-            return MultiNestedTensor(num_rows=1, num_cols=self.num_cols,
-                                     values=values, offset=offset)
-        elif isinstance(index, Tensor) and index.ndim == 1:
-            return self.index_select(index, dim=0)
-        elif isinstance(index, List):
-            return self.index_select(
-                torch.tensor(index, device=self.values.device), dim=0)
+            if isinstance(index[0], int) and isinstance(index[1], int):
+                # Returns Tensor
+                return MultiNestedTensor.get_tensor(self, index[0], index[1])
+            else:
+                # Returns MultiNestedTensor
+                out = self
+                for dim, idx in enumerate(index):
+                    out = MultiNestedTensor.index(out, idx, dim)
+                return out
         else:
-            raise RuntimeError("Advanced indexing not supported yet.")
-
-    def __len__(self):
-        return self.num_rows
-
-    def index_select(self, index: Tensor, dim: int) -> Tensor:
-        if dim == 0:
-            return self._row_index_select(index)
+            # Returns MultiNestedTensor
+            return MultiNestedTensor.index(self, index, dim=0)
+
+    def slice(self, slice: slice, dim: int) -> 'MultiNestedTensor':
+        if dim == 0 or dim == -3:
+            return self.row_slice(slice)
+        elif dim == 1 or dim == -2:
+            return self.col_slice(slice)
```

Comment:
Can we add an assert here in case `dim` is not well defined?

Reply:
We already raise RuntimeError.